### ISSUE TITLE
Issues Deploying

### ISSUE THREAD

**Issue by tobross** [hoppula/refire-forum#25]
I am running into an issue wherein running "npm run deploy" creates error 410 exclaiming that my firebase CLI must be out of date. 

It is not.

I have fully updated the firebase-tools and node, as well as trying version 5.1.1 like some users have suggested. "Firebase deploy --only" commands are depricated so I am only running the base "firebase deploy". I have deleted and reinstalled the modules as well. 

Forgive me for how I've written this. I've been hacking at it for a few days now and haven't gotten it working on the deployed version yet.  It works perfectly on the localhost, however.

Any advice or leads to chase down would be most appreciated.

**Below is the error in question**

=== Deploying to 'necroreact'...

i  deploying database, hosting
i  database: checking rules syntax...
+  database: rules syntax for database necroreact is valid
i  hosting: preparing dist directory for upload...

Error: HTTP Error: 410, This version of the Firebase CLI is no longer able to deploy to Firebase Hosting. Please upgrade to a newer version (>= 4.1.0). If you have further questions, please reach out to Firebase support.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! refire-forum@1.5.0 deploy: `firebase deploy`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the refire-forum@1.5.0 deploy script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\tobro\AppData\Roaming\npm-cache\_logs\2019-02-02T01_50_34_962Z-debug.log

**Comment by hoppula** [comment]
Hi. Thanks for the detailed issue, have you checked if updating bundled `firebase-tools` helps? This project has devDependency of `firebase-tools` version `^3.1.0` and I think running `npm run deploy` calls the local dependency instead of your global, newer version. 

So I'd try to first replace `"firebase-tools": "^3.1.0",` with `"firebase-tools": "^5.0.0",` in `package.json`. There are some API changes in newer `firebase-tools` as you said, so the actual commands in `package.json`'s `deploy:hosting` and `deploy:rules` sections might have to be changed as well.

**Comment by tobross** [comment]
Alright! So, I did what you asked, and then while updating that I found npm-check-updates which is able to rewrite the package.json to to the most recent version of all dependencies. I ran that, and then installed again. I'm not sure whether it was your fix or the other that helped, but the deploy command ran after that.

THAT BEING SAID... Once I deployed, I get a white screen and after investigation, it looks like bundle.js' contents are the same as index.html error below.

Uncaught SyntaxError: Unexpected token < bundle.js:1

When I investigated the sources tab, bundle.js looked exactly like index.html. The only information I had to change in package.json was the html script. I am on a windows machine, so, I had to rewrite it to "html": "mkdir dist && cp index.html dist/index.html" because the dist folder was not being created.

Apologies for having a secondary issue. Still a deployment issue though! Thank you for your assistance!
